cfpb/cfgov-refresh
# Always fall back to CFGOVImage's alt text if the ImageBasic block has no alt value

In instances where we are using the `ImageBasic` block to choose an image for a module, we were not falling back to the alt text set on the `CFGOVImage` itself in the image manager if the `ImageBasic` had no value in its `alt` field. Content editors were confused by this, expecting that the `CFGOVImage` `alt` would be used if one was not set on the `ImageBasic` block. See: GHE/CFGOV/platform/issues/694

This PR changes the behavior of each of those blocks to meet that expectation.

Thanks to @chosak for pairing with me on this!


## Additions

- A new Jinja2 tag, `image_alt_value(ImageBasic)` that takes an `ImageBasic` block and returns the block's `alt` value, if it exits, or the chosen `CFGOVImage`'s `alt` value, if it exists, or an empty string, if neither exist.
- Tests for that tag


## Changes

- Module templates that render ImageBasic blocks use the new `image_alt_value` tag.
- We now override Wagtail's default alt text of an image filename to be the value of the CFGOVImage's `alt` field. In `images.py`, the first addition overrides this for image insertion in rich text fields, and the second addition overrides this for instances of calling them directly, enabling the simplifications in `full-width-text.html` and `post-preview.html`.


## Testing

1. Pull this branch
1. Run the server
1. Upload an image and put something in its `alt` field
1. Add that image to an FCM, Image Inset, Image & Text, Full Width Media, Post Preview, or Snippet List, without setting the `alt` field on the block
1. View the page and see that it has output the `alt` field from the image itself
1. Test other combinations of setting or not setting both alt fields 


## Notes

- In preparing this work, I noticed that [`posts-paginated.html`](https://github.com/cfpb/cfgov-refresh/blob/master/cfgov/jinja2/v1/_includes/posts-paginated.html) and [`post-macros.html`](https://github.com/cfpb/cfgov-refresh/blob/master/cfgov/jinja2/v1/_includes/post-macros.html) no longer seem to be used. Wasn't sure if removing them should be part of this PR or a new one.


## Checklist
